core/linux-dist/evdev.cpp

```diff
@@ -158,8 +158,11 @@
 		ConfigFile mf;
 		mf.parse(fd);
 
+		string name = mf.get("emulator", "mapping_name", "<Unknown>");
+		char * cstr = new char[name.length()+1];
+  		strcpy (cstr, name.c_str());
 		EvdevControllerMapping mapping = {
-			mf.get("emulator", "mapping_name", "<Unknown>").c_str(),
+			cstr,
```

Comment:
I think the right way to fix this is to use a `string` in `EvdevControllerMapping` instead of `char*`

Reply:
https://github.com/reicast/reicast-emulator/blob/d8a7212d69929277d2c4ab54851fb7e97604ec3e/core/linux-dist/evdev.h#L7 make that `const string name`

Reply:
I tried that first, didn't work however.
Might give that a second try, but I'm away for about a week now.

Reply:
Any updates on this?

Reply:
I tested this again, and it's not working.

Changing the header to ``const string name;`` and ``load_mapping()`` to ``mf.get("emulator", "mapping_name", "<Unknown>"),`` (without ``.c_str()``) I get the following output:
``evdev: Using 'P�T%V' mapping``

So it's not working.
Also, changing the code to
````c++
		string name = mf.get("emulator", "mapping_name", "<Unknown>");
		EvdevControllerMapping mapping = {
			name,
````
doesn't work either.